lib/cw-core/src/util.rs

```diff
@@ -57,3 +57,323 @@ pub fn split_framebuffer(framebuffer: &[u8]) -> Vec<Vec<u8>> {
+            //arrange
+            let previous_framebuffer: &mut [u8] = &mut [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4];
+            let current_framebuffer: &mut [u8] = &mut [2, 2, 2, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4];
+
+            let expected_different_framebuffer = vec![0, 0, 2, 2, 2];
+
+            //action
+            let different_framebuffer =
+                get_different_framebuffer(previous_framebuffer, current_framebuffer);
+
+            assert_eq!(
+                different_framebuffer, expected_different_framebuffer,
+                "the result_different_frambuffer is wrong: {:?}",
+                different_framebuffer
+            );
+        }
+
+        #[test]
+        fn test_diff_framebuffer_the_first_and_last_pixel_are_updated_case() {
+            //arrange
+            let previous_framebuffer: &mut [u8] = &mut [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4];
+            let current_framebuffer: &mut [u8] = &mut [2, 2, 2, 1, 1, 1, 2, 2, 2, 3, 3, 3, 9, 9, 9];
+
+            let expected_different_framebuffer = vec![0, 0, 2, 2, 2, 0, 4, 9, 9, 9];
+
+            //action
+            let different_framebuffer =
+                get_different_framebuffer(previous_framebuffer, current_framebuffer);
+
+            //assert
+            assert_eq!(
+                different_framebuffer, expected_different_framebuffer,
+                "the result_different_frambuffer is wrong: {:?}",
+                different_framebuffer
+            );
+        }
+
+        #[test]
+        fn test_diff_framebuffer_incomparable_buffers_case() {
+            //arrange
+            let previous_framebuffer: &mut [u8] = &mut [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4];
+            let current_framebuffer: &mut [u8] = &mut [2, 2, 2, 1, 1, 1, 2, 2, 2, 3, 3, 3];
+
+            let expected_different_framebuffer = vec![0];
+
+            //action
+            let different_framebuffer =
+                get_different_framebuffer(previous_framebuffer, current_framebuffer);
+
+            //assert
+            assert_eq!(
+                different_framebuffer, expected_different_framebuffer,
+                "the result_different_frambuffer is wrong: {:?}",
+                different_framebuffer
+            );
+        }
+
+        #[test]
+        fn test_diff_framebuffer_invalid_buffers_case() {
+            //arrange
+            let previous_framebuffer: &mut [u8] = &mut [0, 0, 0, 1];
+            let current_framebuffer: &mut [u8] = &mut [2, 2, 2, 1];
+
+            let expected_different_framebuffer = vec![0];
+
+            //action
+            let different_framebuffer =
+                get_different_framebuffer(previous_framebuffer, current_framebuffer);
+
+            //assert
+            assert_eq!(
+                different_framebuffer, expected_different_framebuffer,
+                "the result_different_frambuffer is wrong: {:?}",
+                different_framebuffer
+            );
+        }
+    }
```

Comment:
Because the `get_different_framebuffer` function signature will change, the framebuffer will always need to be 1200 long.